### DISCUSSION TITLE
Wrong architecture when linking

### DISCUSSION THREAD

**Discussion by bitdriftr** [mesonbuild/meson#13785] [Q&A]
I am trying to compile a C++ project with Meson on my M1 mac (arm64). However when building I get the following linker error : 

```
ld: warning: ignoring file '/opt/homebrew/Cellar/glib/2.82.1/lib/libglib-2.0.0.dylib': found architecture 'arm64', required architecture 'x86_64'
```

This means the project's object files were built for x86_64, while my system libraries are arm64 as expected. I tried many different solutions (including cross file), but I couldn't find a way to force Meson to build for arm64 instead of x86_64.

Did I miss something? Thanks.

**Comment by eli-schwartz** [comment]
> I tried many different solutions (including cross file), but I couldn't find a way to force Meson to build for arm64 instead of x86_64.
> 
> Did I miss something? Thanks.

Difficult to tell without a description of what you tried.

$STANDARD_LITANY: Can you post e.g. a standalone reproducer, or a link to an existing open source project with the exact setup command plus cross file contents as applicable?
